.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+fail_fast: false
+repos:
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: master
+    hooks:
+      - id: clang-format
+        args:
+          - -i
```

Comment:
Does this pre-commit format and stage updates and include them in your commit? Or does this generate a new set of changes and bail, letting you verify first?

Reply:
the latter.

The commit fails (it's pre-commit hook after all) and then you need to add the changed files and commit again. So it's impossible to commit badly formatted files, except by using the `-n` flag with `git commit`.